app/src/main/java/com/valhalla/thor/presentation/launcher/FreezerLaunchActivity.kt

```
@@ -68,14 +73,56 @@ class FreezerLaunchActivity : Activity() {
+            // translucent but *resumed*, so it is foreground and its toast always renders. For
+            // Unfreeze-all that matters: the notification needs permission and the QS tile is
+            // freeze-only, so with notifications off an unfreeze used to report nothing at all.
+            //
+            // The wait is bounded well inside the run's own 30s deadline: this window is
+            // invisible but touchable, so sitting on it would swallow taps meant for the
+            // launcher underneath. Past the window we acknowledge and get out of the way — the
+            // run belongs to the app scope and finishes (and notifies) without us.
+            val result = withTimeoutOrNull(REPORT_WINDOW_MS) {
+                try {
+                    run.await()
+                } catch (e: CancellationException) {
```

detekt (1.23.8)

[warning] The caught exception is swallowed. The original exception could be lost.

(detekt.exceptions.SwallowedException)